addons/web_editor/static/src/js/wysiwyg/widgets/link.js

```diff
@@ -80,11 +80,10 @@ const Link = Widget.extend({
                 }
                 $node = $node.parent();
             }
-            const textLink = this.$link[0] || this.data.range.cloneContents();
-            const [encodedText, images] = this._encodeNodeToText(textLink);
-            this.data.originalText = this._decodeText(encodedText, images);
-            this.data.content = encodedText;
-            this.data.images = images;
+            const linkNode = this.$link[0] || this.data.range.cloneContents();
+            const linkText = linkNode.innerHTML || linkNode.textContent;
```

Comment:
How can we have `innerHTML` falsy but `textContent` not?

Reply:
If you select only some text from a TextNode, then `this.data.range.cloneContents()` returns a DocumentFragment that only contains a `TextNode` and `innerHTML` is `undefined` in that case.

Reply:
:+1: